src/ansys/fluent/core/solver/flobject.py

```diff
@@ -595,7 +595,14 @@ def get_state(self) -> StateT:
     def set_state(self, state: Optional[StateT] = None, **kwargs):
         """Set the state of the object."""
         with self.while_setting_state():
-            return self.flproxy.set_var(self.path, self.to_scheme_keys(kwargs or state))
+            if isinstance(state, (tuple, ansys_units.Quantity)) and hasattr(
```

Comment:
We get an exception for tuple, so that case is catchable, but for `Quantity` the code falls through to defaulting to string and the object is implicitly convertible, so no exception is catchable - we just end up sending a rubbish expression. Hence this hack.